cli/src/semgrep/app/scans.py

```diff
@@ -279,35 +281,18 @@ def report_failure(self, exit_code: int) -> None:
         except requests.RequestException:
             raise Exception(f"API server returned this error: {response.text}")
 
-    def report_findings(
+    def _build_ci_scan_results(
         self,
-        matches_by_rule: RuleMatchMap,
-        errors: List[SemgrepError],
         rules: List[Rule],
         targets: Set[Path],
         renamed_targets: Set[Path],
-        ignored_targets: FrozenSet[Path],
-        parse_rate: ParsingData,
-        total_time: float,
         commit_date: str,
         lockfile_dependencies: Dict[str, List[out.FoundDependency]],
-        dependency_parser_errors: List[DependencyParserError],
         contributions: out.Contributions,
-        engine_requested: "EngineType",
-        progress_bar: "Progress",
-    ) -> ScanCompleteResult:
-        """
-        commit_date here for legacy reasons. epoch time of latest commit
-
-        Returns (success, block_scan, block_reason)
-        """
-        state = get_state()
+        all_matches: list[RuleMatch],
```

Comment:
This is causing the CI failure. Some test is using an older version of python where this isn't valid. Use `List` instead of `list`